Add Support for Appearance Tools

danielwrobert opened the pull request:
This provides support for spacing, border, typography, and color options, even without using `theme.json`.

Properties that should be added to various block settings are:
- [Border](https://developer.wordpress.org/themes/global-settings-and-styles/border)
- [Color](https://developer.wordpress.org/themes/global-settings-and-styles/color)
- [Dimensions](https://developer.wordpress.org/themes/global-settings-and-styles/dimensions)
- [Position](https://developer.wordpress.org/themes/global-settings-and-styles/position)
- [Spacing](https://developer.wordpress.org/themes/global-settings-and-styles/spacing)
- [Typography](https://developer.wordpress.org/themes/global-settings-and-styles/typography)

Additional references:
- https://wordpress.org/documentation/wordpress-version/version-6-5/#add-appearance-tools-to-classic-themes
- https://developer.wordpress.org/themes/global-settings-and-styles/settings/appearance-tools/

<!-- Briefly describe the issue or problem that this PR solves. -->

<!-- Explain your fix - how it addresses the problem, what else might be affected, any risks etc. -->

<!-- Don't forget to update the title with something descriptive. -->

### Screenshots
<!-- If your change has a visual component, add a screenshot here. A "before" screenshot would also be helpful. -->
#### Image Block
<table>
<tr>
	<td> Before
	<td> After
<tr>
	<td> <img src="https://github.com/woocommerce/storefront/assets/481776/abc4f64e-202e-4753-86f0-f4e096d16ebb" />
	<td> <img src="https://github.com/woocommerce/storefront/assets/481776/3407d51b-6567-4884-acae-8fa08e0e8019">
</table>

#### Text Block
<table>
<tr>
	<td> Before
	<td> After
<tr>
	<td> <img src="https://github.com/woocommerce/storefront/assets/481776/9e3b98fb-9750-4e23-b003-4cc575801352" />
	<td> <img src="https://github.com/woocommerce/storefront/assets/481776/c43f21c8-ba83-44f3-a394-c8197cef33ee">
</table>

#### Layout Block
<table>
<tr>
	<td> Before
	<td> After
<tr>
	<td> <img src="https://github.com/woocommerce/storefront/assets/481776/154f3359-bbe3-4f9e-86db-78de0301e319" />
	<td> <img src="https://github.com/woocommerce/storefront/assets/481776/ec6d7eaf-a75d-4818-85dc-286cee7db5bd">
</table>

### How to test the changes in this Pull Request:
<!-- Add detailed instructions for how to test that this PR fixes the issue, and confirm that it doesn't break any other features :) -->

1. Insert an Image block in the editor and confirm that the sidebar settings have an additional tab added for Styles (see the **After** screenshot under **Image Block** above). Confirm changes to the new settings are reflected in the editor and on the front end.
2. Insert a text block (Heading, Paragraph, etc) in the editor and confirm the Dimensions section added to the sidebar settings (see the **After** screenshot under **Text Block** above). This should allow additional spacing options (namely Margin). Confirm changes to the new settings are reflected in the editor and on the front end.
3. Insert a layout block (Row or Column) in the editor and confirm added controls under the Styles tab in the sidebar settings(see the **After** screenshot under **Layout Block** above). Confirm changes to the new settings are reflected in the editor and on the front end.
4. Smoke test additional Core blocks to ensure the new properties are added where expected and do not have any unintended side-effects.

### Changelog

<!-- Add suggested changelog entry here. For example: -->

> Add – support for Appearance Tools in Classic themes. #2143

<!-- See [previous releases](https://github.com/woocommerce/storefront/releases) for more examples. -->

danielwrobert committed:
Add support for appearance tools.

This provides support for spacing, border, typography, and color
options, even without using theme.json

See: https://wordpress.org/documentation/wordpress-version/version-6-5/#add-appearance-tools-to-classic-themes

tjcafferkey reviewed (commented):
The change tests well and looks good to me. Just wondering if the `languages/storefront.pot` file update is needed?

danielwrobert commented:
> The change tests well and looks good to me. Just wondering if the `languages/storefront.pot` file update is needed?

Good point. While we're tracking it and should keep it up to date, it doesn't need to be included here.

I've removed that commit from this PR.

I also rebased with `trunk` and I see @gigitux already updated the POT file today so we're all set there!
